### PR TITLE
fix (combinations n k) bug

### PR DESCRIPTION
**Pull request by bennn**
Now using Gosper's hack to enumerate length k binary numbers.
New implementation is shorter & a little more obviously correct
(if you trust the bit-twiddling)
https://en.wikipedia.org/wiki/Combinatorial_number_system#Applications

### CODE REVIEW

**Comment by samth** [comment]
The link to Wikipedia should be in the source code.

**Comment by bennn** [comment]
Added [wiki link](https://github.com/racket/racket/pull/1242/files#diff-607ab65a73af9b2046acf4cfbb443282R624)

File: racket/collects/racket/list.rkt [hunk 1]
```diff
@@ -604,69 +604,40 @@
...
+     [else
+      ;; Enumerate numbers with `k` ones, smallest to largest
+      (define first (- (expt 2 k) 1))
+      (define gospers-hack ;; https://en.wikipedia.org/wiki/Combinatorial_number_system#Applications
+        (if (zero? first)
+          add1
+          (lambda (n)
+            (let* ([u (bitwise-and n (- n))]
+                   [v (+ u n)])
+              (+ v (arithmetic-shift (/ (bitwise-xor v n) u) -2))))))
```
Comment (stamourv) [inline racket/collects/racket/list.rkt]: Should this `/` be `quotient` to ensure the same behavior as that Wikipedia code?

**Comment by stamourv** [comment]
LGTM
Please close PR15251 when you push.

**Comment by bennn** [comment]
merged :tada:

**Pull request closed without merging**